pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/AutoConsumeSchema.java

```diff
@@ -43,73 +44,82 @@
 @Slf4j
 public class AutoConsumeSchema implements Schema<GenericRecord> {
 
-    private Schema<?> schema;
+    private final Map<SchemaVersion, Schema<?>> schemaMap = new HashMap<>();
```

Comment:
this should be a ConcurrentHashMap because the Schema can be accessed by multiple threads concurrently
for instance a Consumer that consumes more messages and process them in parallel, calling `Message.getReaderSchema() `or `Message.getValue()`

Reply:
Yes, I'll fix.